compiler/rustc_resolve/src/macros.rs

```diff
@@ -1076,10 +1077,22 @@ impl<'a> Resolver<'a> {
 
         if result.is_builtin {
             // The macro was marked with `#[rustc_builtin_macro]`.
-            if let Some(ext) = self.builtin_macros.remove(&item.ident.name) {
+            if let Some(builtin_macro) = self.builtin_macros.get_mut(&item.ident.name) {
+                use crate::BuiltinMacroState::*;
+
                 // The macro is a built-in, replace its expander function
                 // while still taking everything else from the source code.
-                result.kind = ext.kind;
+                // If we already loaded this builtin macro, give a better error message than 'no such builtin macro'.
+                match mem::replace(builtin_macro, AlreadySeen(item.span)) {
+                    NotYetSeen(ext) => result.kind = ext.kind,
+                    AlreadySeen(span) => {
+                        let msg = "attempted to define built-in macro more than once";
+                        self.session
+                            .struct_span_err_with_code(item.span, msg, error_code!(E0773))
+                            .span_note(span, "previously defined here")
```

Comment:
In practice the current span and the previous span are going to point to the same place though, if we are talking about issues like #75176.

Reply:
For https://github.com/rust-lang/rust/pull/75176, it would have been the same, yes. However that in itself would have been helpful debugging the issue: it shows that the same crate is being loaded twice. And for problems that are less obscure this will be very helpful since it points straight to the other location that conflicts.